src/libopenrave-core/colladaparser/colladareader.cpp

```diff
@@ -988,6 +1004,12 @@ class ColladaReader : public daeErrorHandler
         if( bSuccess && _prefix.size() > 0 ) {
             _AddPrefixForKinBody(pbody,_prefix);
         }
+
+        if( bSuccess ) {
+            // for extracting a single body from file, need to assign a random body id to not conflict with what's already in environment
+            pbody->_id = str(boost::format("body%d")%utils::GetNanoTime());
```

Comment:
@ziyan same as above

Reply:
For now, I think empty id is okay for this case. Resolved.